### ISSUE TITLE
Spa default page should not intercept requests for api paths

### ISSUE THREAD

**Issue by GeekyMonkey** [aspnet/JavaScriptServices#1463]
Everything works fine on the dev machine.  But when I publish, then requests for api data in /api/mydata/mystuff
Get intercepted by the spa middleware.

If it can't be changed to respect the MVC routes that have been set-up.  Perhaps the middleware could have an option with regex paths to ignore.  Something like this in SpaDefaultPageMiddleware.cs

            // Rewrite all requests to the default page 
            app.Use((context, next) => 
            { 
                if (!options.IgnorePaths.Any((p) => { return Regex.IsMatch(context.Request.Path, p); })) 
                { 
                    context.Request.Path = options.DefaultPage; 
                } 
                return next(); 
            }); 

Or is there some other way to solve this problem?

**Comment by AnshulKhandelwal02** [comment]
I faced the same issue when moved from Webpack template to new Angular CLI template. I had serilog integrated to my project, wherein /log route was MVC route used to show logs. But post migrating to CLI template, system is trying to find this route in client app.
Previously I thought there is some configuration issue in my code, and I didn't migrate correctly. 
But seeing your comment, I think it's the new template's issue.

Any way to solve this problem ?

**Comment by GeekyMonkey** [comment]
Turns out my problem was a red herring.  So long as you have the MVC and static routes configured before the Spa, then those should be getting first attention.  I was testing in Chrome, and the requests to my api were not being hit.  Out of desperation, I tried IE, and it WORKED there!  The difference is that IE doesn't support the service worker, and therein was my problem.  Adding the following to ngsw-config.json solved my problem (after clearing the cache in Chrome)

  "dataGroups": [
    {
      "name": "api",
      "urls": [ "/api/**" ],
      "cacheConfig": {
        "maxSize": 0,
        "maxAge": "0u",
        "strategy": "freshness"
      }
    }
